File: zipopener.java
import net.lingala.zip4j.ZipFile;
import net.lingala.zip4j.exception.ZipException;
import net.lingala.zip4j.model.FileHeaders;
import net.lingala.zip4j.model.ZipParameters;

import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;

public class ZipCracker {

    public static void main(String[] args) {
        String zipFilePath = "protected.zip";  // Path to your ZIP file
        String passwordList = "passwords.txt"; // Path to your password dictionary file

        try {
            crackZip(zipFilePath, passwordList);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    public static void crackZip(String zipFilePath, String passwordList) throws IOException {
        ZipFile zipFile = new ZipFile(zipFilePath);

        try (BufferedReader reader = new BufferedReader(new FileReader(passwordList))) {
            String password;
            while ((password = reader.readLine()) != null) {
                password = password.trim();
                if (tryPassword(zipFile, password)) {
                    System.out.println("Password found: " + password);
                    return;
                }
            }
            System.out.println("Password not found.");
        }
    }

    public static boolean tryPassword(ZipFile zipFile, String password) {
        try {
            if (zipFile.isEncrypted()) {
                zipFile.setPassword(password.toCharArray());
            }
            // Try to extract to check if the password is correct
            zipFile.extractAll("output_folder");  // Extract to a folder
            return true;
        } catch (ZipException e) {
            // Wrong password or issue with extraction
            return false;
        }
    }
}

File: zipopener.py
import zipfile

def crack_zip(zip_file, password_list):
    with zipfile.ZipFile(zip_file, 'r') as zfile:
        with open(password_list, 'r') as passwords:
            for password in passwords:
                password = password.strip()
                try:
                    zfile.extractall(pwd=password.encode('utf-8'))
                    print(f"Password found: {password}")
                    return True
                except (RuntimeError, zipfile.BadZipFile):
                    # Incorrect password or bad zip file
                    continue
    print("Password not found.")
    return False

# Usage example:
zip_file = 'protected.zip'  # Path to your zip file
password_list = 'passwords.txt'  # Path to your password dictionary file
crack_zip(zip_file, password_list)
